src/main/java/org/jabref/gui/frame/JabRefFrameViewModel.java

```diff
@@ -252,7 +258,7 @@ private void openDatabases(List<ParserResult> parserResults) {
         // if we found new entry types that can be imported, or checking
         // if the database contents should be modified due to new features
         // in this version of JabRef.
-        parserResults.forEach(pr -> OpenDatabaseAction.performPostOpenActions(pr, dialogService));
+        parserResults.forEach(pr -> OpenDatabaseAction.performPostOpenActions(pr, dialogService, prefs));
```

Comment:
Strange that the whole preferences are sent. Can't the injector be used inside `performPostOpenActions`?

Reply:
Using the Injector class the same way as the infamous Globals class wouldn't be much progress.

Reply:
Injection is about having a global "store" of available services. Injection is about reducing code (here: passing variables). 😅

Reply:
Please let's work on that in a follow up PR. There are hundreds of code snippets like this that would have to be altered if we want to make this pr perfect, but many of these snippets would have to be touched again when implementing a proper di framework (at least if we decide for constructor injection, which is in most di frameworks the preferred way, since its easier to test.)

We can decide to work for weeks and months an this pr, but it would stall everybody else who depends on this pr merged.

Rome wasn't build in one day either.